pkg/services/sqlstore/migrations/ualert/tables.go

```diff
@@ -264,4 +264,14 @@ func AddAlertmanagerConfigMigrations(mg *migrator.Migrator) {
 
 	mg.AddMigration("alert alert_configuration alertmanager_configuration column from TEXT to MEDIUMTEXT if mysql", migrator.NewRawSQLMigration("").
 		Mysql("ALTER TABLE alert_configuration MODIFY alertmanager_configuration MEDIUMTEXT;"))
+
+	// add org_id column; set default to 1 (Main org.)
+	mg.AddMigration("add column org_id in alert_configuration", migrator.NewAddColumnMigration(alertConfiguration, &migrator.Column{
+		Name: "org_id", Type: migrator.DB_BigInt, Nullable: false, Default: "0",
```

Comment:
Setting default to zero has the implication that configurations created by the migration before this change will become invisible to the users (users have to re-run the migration by disabling-enabling the `ngalert` feature) to get their configurations back.

However I prefer this than setting the default to the 1 (=the main organisation ID).

Any suggestions for handling better this are welcome!

Reply:
Alerting objects (contact points, policies) may have been created outside the dashboard->ngalert migration after the enabling of the feature toggle. The proper org for these would be unknown. Therefore all config must be cleared with the adding of Orgs for things to be in a valid state. If this is correct, then the main issue I think is: Does Grafana take care of this, or does a user have to do it as part of the upgrade? 

It seems more desirable to me for the user (or some subset) not have to do this. If it is not required that a user do it, then it seems like it should be an additional migration entry. However, since everything needs to be cleared anyways, I think it is maybe okay to modify the migration code and point to the same code with two entries. I believe this would effectively make it so the migration is done twice going forward though, which is perhaps not good.

Reply:
I agree that all Alertmanager configuration should be cleared (since their organisation is unknown) and I like the idea of doing it automatically without the users to take any extra action. But I think that we need to do it in a way that users will not loose also the rules they have created as well.

Reply:
I've left a comment for this on Slack, let me know if you want me to expand here.

Reply:
We have decided to force re-running the migration automatically without having the users to do anything.